linker/symbols.go

```diff
@@ -232,29 +230,29 @@ func (s *packageSymbols) importPackage(pkgSpan ast.SourceSpan, pkg protoreflect.
 	return child, nil
 }
 
-func (s *Symbols) getPackage(pkg protoreflect.FullName) *packageSymbols {
+func (s *Symbols) getPackage(pkg protoreflect.FullName, exact bool) *packageSymbols {
 	if pkg == "" {
 		return &s.pkgTrie
 	}
-
-	parts := strings.Split(string(pkg), ".")
-	for i := 1; i < len(parts); i++ {
-		parts[i] = parts[i-1] + "." + parts[i]
-	}
-
 	cur := &s.pkgTrie
-	for _, p := range parts {
+	enumerator := nameEnumerator{name: pkg}
+	for {
```

Comment:
Could the lookup be reversed? Is it needed to go from `a` -> `a.b` -> `a.b.c` and not just lookup `a.b.c` directly?
```go
for ; pkg != ""; pkg = pkg.Parent() {
    cur.mu.RLock()
    next := cur.children[p]
    cur.mu.RUnlock()
    if next != nil || exact {
        return next
    }
}
return &s.pkgTrie
```